### PR TITLE
ci: release workflow

### PR DESCRIPTION
**Pull request by trim21**

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  - pull_request
+
+jobs:
+  build:
+    name: Compile
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: brndnmtthws/rust-action-rustup@v1
```
Comment (timofurrer) [inline .github/workflows/build.yaml:14]: What's the reason for using this and not `actions-rs/toolchain@v1` ?

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        with:
+          toolchain: stable
+      - uses: Swatinem/rust-cache@v2
```
Comment (timofurrer) [inline .github/workflows/build.yaml]: It's not a huge project and won't be built often, so I'd rather only rely on what is really necessary:

```suggestion

```

File: .github/workflows/build.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      - uses: actions-rs/cargo@v1
+        with:
+          command: build
+          args: --release --bin retry
+        env:
+          CARGO_TARGET_DIR: "./dist/"
+
```
Comment (timofurrer) [inline .github/workflows/build.yaml]: I think we don't really care about the directory, so `target` is good enough?

```suggestion
```

File: .github/workflows/build.yaml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+      - uses: actions/upload-artifact@v3
+        with:
+          name: binary
+          path: dist/release/retry
```
Comment (timofurrer) [inline .github/workflows/build.yaml]: Goes with the above suggestion to leave it at `target`:

```suggestion
          path: target/release/retry

```

File: .github/workflows/release.yaml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    steps:
+      - name: Get Tag Name
+        run: echo "TAG=${GITHUB_REF##*/}" >> $GITHUB_ENV
+
+      - name: Upload Github Release
+        run: gh release create "$TAG"
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  release:
```
Comment (timofurrer) [inline .github/workflows/release.yaml:25]: Please apply the same suggestions as the ones I left for the `build` workflow.

**Review by timofurrer — CHANGES_REQUESTED** [review]
@trim21 already looks pretty solid. 

I've left a few comments and back to you 🏓

File: .github/workflows/build.yaml (hunk 1 above)
Comment (trim21) [reply .github/workflows/build.yaml:14]: it hasn't update since 2020 and still using deprecsted `set-output` command

**Comment by timofurrer** [comment]
LGTM! Thanks @trim21 !

**Comment by trim21** [comment]
new tag please?

**Comment by timofurrer** [comment]
@trim21 there is a [v1.0.1](https://github.com/timofurrer/retry-cmd/releases/tag/v1.0.1) release now!

**Pull request merged**